openquake/commonlib/tests/source_test.py

```diff
@@ -477,6 +479,65 @@ def test_raises_useful_error_2(self):
             "node areaSource: No subnode named 'nodalPlaneDist'"
             " found in 'areaSource', line 5 of", str(ctx.exception))
 
+    def test_hypolist_but_not_sliplist(self):
+        simple_file = StringIO("""\
+<?xml version='1.0' encoding='utf-8'?>
+<nrml xmlns:gml="http://www.opengis.net/gml"
+      xmlns="http://openquake.org/xmlns/nrml/0.4">
+    <sourceModel name="Some Source Model">
+        <simpleFaultSource
+        id="3"
+        name="Mount Diablo Thrust"
+        tectonicRegion="Active Shallow Crust"
+        >
+            <simpleFaultGeometry>
+                <gml:LineString>
+                    <gml:posList>
+                        -121.8229 37.7301 -122.0388 37.8771
+                    </gml:posList>
+                </gml:LineString>
+                <dip>
+                    45.0
+                </dip>
+                <upperSeismoDepth>
+                    10.0
+                </upperSeismoDepth>
+                <lowerSeismoDepth>
+                    20.0
+                </lowerSeismoDepth>
+            </simpleFaultGeometry>
+            <magScaleRel>
+                WC1994
+            </magScaleRel>
+            <ruptAspectRatio>
+                1.5
+            </ruptAspectRatio>
+            <incrementalMFD
+            binWidth="0.1"
+            minMag="5.0"
+            >
+                <occurRates>
+                    0.0010614989 0.00088291627 0.00073437777 0.0006108288 0.0005080653
+                </occurRates>
+            </incrementalMFD>
+            <rake>
+                30.0
+            </rake>
+            <hypoList>
+                <hypo alongStrike="0.25" downDip="0.25" weight="0.3"/>
+                <hypo alongStrike="0.75" downDip="0.75" weight="0.7"/>
+            </hypoList>
+        </simpleFaultSource>
+    </sourceModel>
+</nrml>
+""")
+        # check that the error raised by hazardlib is wrapped correctly
```

Comment:
Ciao, Marco, yes, I agree. It should be a better idea not to make the hypocentre distribution and slip distribution explicitly used for directivity calculation. Maybe tomorrow we can discuss how to put this test for directivity calculation.